frontend/src/utilities/__tests__/useDeepCompareMemoize.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { useDeepCompareMemoize } from '~/utilities/useDeepCompareMemoize';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+
+describe('useDeepCompareMemoize', () => {
+  it('should return a memoized version of the input value ', () => {
+    const initialData = {
+      config: {
+        theme: 'dark',
+        language: 'english',
+        fontSize: 16,
+      },
+    };
+
+    const newData = {
+      config: {
+        theme: 'light',
+        language: 'french',
+        fontSize: 18,
+      },
+    };
+
+    const identicalData = {
+      config: {
+        theme: 'dark',
+        language: 'english',
+        fontSize: 16,
+      },
+    };
+    expect(testHook(useDeepCompareMemoize)(initialData).result.current).toBe(initialData);
+
+    // Re-render with new data, expect result to change
+    expect(testHook(useDeepCompareMemoize)(newData).result.current).toBe(newData);
```

Comment:
What you've done is not a re-render but a brand new render.
`testHook` returns a `rerender` function that should be used here.

Please see `hooks.spec.ts` for examples on `rerender`.